python/test/result_test.py

```diff
@@ -244,5 +249,115 @@ def test_load_result_with_none(self):
         result.set_score_aggregate_method(ListStats.nonemean)
         self.assertAlmostEqual(result['STRRED_feature_srred_score'], 5829.2644469999996, places=4)
 
+class ResultAggregatingTest(unittest.TestCase):
+
+    def test_invalid_format_aggregator(self):
+
+        print 'test on running result aggregation with invalid format...'
+        with self.assertRaises(AssertionError):
+            XML_Assembler(['dummy.png']).assemble()
+            XML_Assembler(['dummy.png, dummpy.xml']).assemble()
+            JSON_Assembler(['dummy.png']).assemble()
+            JSON_Assembler(['dummy.xml, dummpy.png']).assemble()
+
+    def test_xml_aggregator(self):
+
+        print 'test on running XML aggregation...'
+        ref_path, dis_path, asset, asset_original = set_default_576_324_videos_for_testing()
+
+        asset_list = [asset, asset_original]
+
+        self.runner = VmafQualityRunner(
+            asset_list,
+            None, fifo_mode=True,
+            delete_workdir=True,
+            result_store=None,
+            optional_dict={
+                'model_filepath': VmafConfig.model_path("vmaf_v0.6.1.pkl"),
+            },
+            optional_dict2=None,
+        )
+        self.runner.run()
+
+        results = self.runner.results
+
+        xml_file_list = []
+        dump_dir = os.getcwd() + '/temp_outputs/'
```

Comment:
Do not do this. Instead, use VmafConfig.workdir_path(...). Follow example at:
https://github.com/Netflix/vmaf/blob/master/python/test/command_line_test.py#L16

Also, for tests, each test must be 'memoryless', that means each test should not interfere another. Therefore, you must clean up your temp files after creating them. This is best done in the setUp() and tearDown() function. Follow the example above.